tests/e2e/test_spyre_cb.py

```diff
@@ -83,177 +83,175 @@ def test_cb_handling(
-        assert len(engine_core.scheduler.running) == 2
-        request_outputs = engine.step()
-        assert len(request_outputs) == 2  # requests 2 and 3 decoding now
-        assert request_outputs[0].request_id == "3"  # req 3 is decoding
-        assert request_outputs[1].request_id == "2"  # req 2 is decoding
-
-        assert len(engine_core.scheduler.waiting) == 0
-        assert len(engine_core.scheduler.running) == 2
-        request_outputs = engine.step()
-        assert len(request_outputs) == 2  # requests 2 and 3 decoding now
-        assert request_outputs[0].request_id == "3"  # req 3 is decoding
-        assert request_outputs[1].request_id == "2"  # req 2 is decoding
-        assert request_outputs[1].finished  # request 2 is done
-        assert request_outputs[1].outputs[0].text == " 6 5 4 3 2 "
-
-        assert len(engine_core.scheduler.waiting) == 0
-        assert len(engine_core.scheduler.running) == 1
+    assert len(engine_core.scheduler.waiting) == 1
+    assert len(engine_core.scheduler.running) == 0
+
+    request_outputs = engine.step()
+    assert len(request_outputs) == 1  # only 1 request
+    assert request_outputs[0].request_id == "1"  # req 1 is decoding (prefill step)
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 1
+
+    # add another request
+    engine.add_request("2", prompt2, sampling_params)
+
+    assert len(engine_core.scheduler.waiting) == 1  # req 2 is waiting
+    assert len(engine_core.scheduler.running) == 1
+
+    request_outputs = engine.step()
+    assert len(request_outputs) == 1  # still only 1 request
+    assert request_outputs[0].request_id == "2"  # req 2 is decoding (prefill step)
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # both requests decoding now
+    assert request_outputs[0].request_id == "2"  # req 2 is decoding
+    assert request_outputs[1].request_id == "1"  # req 1 is decoding
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # both requests decoding now
+    assert request_outputs[0].request_id == "2"  # req 2 is decoding
+    assert request_outputs[1].request_id == "1"  # req 1 is decoding
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+
+    request_outputs = engine.step()
+    assert len(request_outputs) == 2  # both requests decoding now
+    assert request_outputs[0].request_id == "2"  # req 2 is decoding
+    assert request_outputs[1].request_id == "1"  # req 1 is decoding
+
+    assert len(engine_core.scheduler.waiting) == 0
+    assert len(engine_core.scheduler.running) == 2
+
+    # add the third request
+    # but since VLLM_SPYRE_MAX_BATCH_SIZE=2
```

Comment:
just realized we don't set this env var anymore
```suggestion
    # but since max_num_seqs=2
```

Reply:
good catch